lib/splitclient-rb/validators.rb

```diff
@@ -38,6 +41,25 @@ def valid_matcher_arguments(args)
       true
     end
 
+    def valid_flag_sets(method, flag_sets)
+      if flag_sets.nil? || !flag_sets.is_a?(Array)
+        return nil
+      end
+      valid_flag_sets = SortedSet[]
+      flag_sets.compact.uniq.select do |flag_set|
+        if !flag_set.is_a?(String)
+          log_invalid_type(:flag_set, method)
+        elsif flag_set.is_a?(String) && flag_set.empty?
+          log_nil(:flag_set, method)
+        elsif !flag_set.empty? && string_match?(flag_set.strip, method)
+          valid_flag_sets.add(flag_set.strip.downcase)
+        else
+          @config.logger.warn("#{method}: you passed an invalid flag set, flag set name must be a non-empty String")
+        end
+      end
+      !valid_flag_sets.empty? ? valid_flag_sets :  nil
```

Comment:
you should return a empty `set` and not `nil` to avoid future failures.

Reply:
will do